File: CORREDOR/Tipos Constructores/JS/Carro.js
const Transporte = require("./Transporte");

//Constructor sin parametros
class Carro extends Transporte {
    constructor() {
        super(); // llama al constructor de Transporte
        this.tipo = "Carro";
        this.marca = "Toyota";
        this.modelo = 2020;
    }

    mover() {
        console.log("El carro está en movimiento.");
    }

    informacion() {
        console.log("Tipo de transporte:", this.tipo);
        console.log("Marca:", this.marca);
        console.log("Modelo:", this.modelo);
    }
}
module.exports = Carro;

const c = new Carro();
c.mover();
c.informacion();

File: CORREDOR/Tipos Constructores/Go/Carro.go
package main

import "fmt"

type Carro struct {
	Transporte
	Marca  string
	Modelo int
}

//Constructor Sin Parametros
func NewCarro() Carro {
	return Carro{
		Transporte: NewTransporte(),
		Marca:      "Toyota",
		Modelo:     2020,
	}
}

func (c Carro) Mover() {
	fmt.Println("El carro está en movimiento.")
}

func (c Carro) Informacion() {
	fmt.Println("Tipo:", c.Tipo)
	fmt.Println("Marca:", c.Marca)
	fmt.Println("Modelo:", c.Modelo)
}